src/core/qgsuserconfigmanager.h

```diff
@@ -0,0 +1,33 @@
+#ifndef QGSUSERCONFIGMANAGER_H
+#define QGSUSERCONFIGMANAGER_H
+
+#include <qgis.h>
+
+class CORE_EXPORT QgsUserConfig
+{
+  public:
+    QgsUserConfig( QString folder );
+    const QString folder() const;
+
+  private:
+    QString mConfigFolder;

```

Comment:
Yeah I was thinking about expanding it out to inclulding stuff like python folder, processing folder, etc.  

Trying to avoid having people work out that path themselves.  Will move all stuff from `QgsApplication` to here if it makes sense.  

Just a thought anyway.